### PR TITLE
Do not monadic-chain with an if-else block

### PR DESCRIPTION
**Pull request by HyukjinKwon**
From https://github.com/apache/spark/pull/37390.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @cloud-fan, @viirya FYI

File: README-KO.md [hunk 1]
```diff
@@ -842,6 +844,25 @@ def getAddress(name: String): Option[String] = {
...
+
+// A more readable approach.
+val ret = if (condition) {
+  Seq(1, 2, 3)
+} else {
+  Seq(1, 2, 3)
+}
+ret.map(_ + 1)  // Results in List(2, 3, 4)
+```
+
```
Comment (HyukjinKwon) [inline README-KO.md:865]: cc @HeartSaVioR too if you happen to find some time to take a look 👍

File: README-KO.md [hunk 2]
```diff
@@ -842,6 +844,25 @@ def getAddress(name: String): Option[String] = {
...
 
+다른 예로는 if-else가 전체 if-else의 모나드 체인인지 아니면 else 절의 모나드 체인인지 햇갈리는 경우가 있습니다.
+```scala
+// A monadic chaining approach
+val condition: Boolean = ...
+if (condition) {
+  Seq(1, 2, 3)  // Results in List(2, 3, 4)
+} else {
+  Seq(1, 2, 3)  // Results in List(1, 2, 3)
+}.map(_ + 1)
```
Comment (HeartSaVioR) [inline README-KO.md:855]: Wait, where the monad chain applies? I have been thinking that it applies to the entire if-else, but if it doesn't, it should be else, then code comment seems to be swapped.

File: README.md [hunk 3]
```diff
@@ -844,6 +846,24 @@ def getAddress(name: String): Option[String] = {
...
 ```
 
+Another example is an if-else block that confuses if the chain is for the whole if-else or only `else` block.
+```scala
+// A monadic chaining approach
+val condition: Boolean = ...
+if (condition) {
+  Seq(1, 2, 3)  // Results in List(2, 3, 4)
+} else {
+  Seq(1, 2, 3)  // Results in List(1, 2, 3)
```
Comment (HeartSaVioR) [inline README.md:856]: Same here.

File: README-KO.md (hunk 2 above)
Comment (HyukjinKwon) [reply README-KO.md:855]: uh oops

File: README-KO.md [hunk 4]
```diff
@@ -842,6 +844,25 @@ def getAddress(name: String): Option[String] = {
 
 ```
 
+다른 예로는 if-else가 전체 if-else의 모나드 체인인지 아니면 else 절의 모나드 체인인지 햇갈리는 경우가 있습니다.
+```scala
+// A monadic chaining approach
+val condition: Boolean = ...
+if (condition) {
+  Seq(1, 2, 3)  // Results in List(2, 3, 4)
```
Comment (HyukjinKwon) [inline README-KO.md:852]:
```suggestion
  Seq(1, 2, 3)  // Results in List(1, 2, 3)
```

File: README-KO.md [hunk 5]
```diff
@@ -842,6 +844,25 @@ def getAddress(name: String): Option[String] = {
...
+} else {
+  Seq(1, 2, 3)  // Results in List(1, 2, 3)
```
Comment (HyukjinKwon) [inline README-KO.md:854]:
```suggestion
  Seq(1, 2, 3)  // Results in List(2, 3, 4)
```

File: README.md [hunk 6]
```diff
@@ -844,6 +846,24 @@ def getAddress(name: String): Option[String] = {
 
 ```
 
+Another example is an if-else block that confuses if the chain is for the whole if-else or only `else` block.
+```scala
+// A monadic chaining approach
+val condition: Boolean = ...
+if (condition) {
+  Seq(1, 2, 3)  // Results in List(2, 3, 4)
```
Comment (HyukjinKwon) [inline README.md:854]:
```suggestion
  Seq(1, 2, 3)  // Results in List(1, 2, 3)
```

File: README.md (hunk 3 above)
Comment (HyukjinKwon) [inline README.md:856]:
```suggestion
  Seq(1, 2, 3)  // Results in List(2, 3, 4)
```

**Review by cloud-fan — APPROVED** [review]

**Review by viirya — COMMENTED** [review]
lgtm

**Review by HeartSaVioR — COMMENTED** [review]
LGTM of course! :)

**Pull request merged**